src/main/java/edu/tamu/sage/job/JobRunner.java

```diff
@@ -0,0 +1,98 @@
+    private static final String DATE_KEY = "date";
+    private static final String DAYS_KEY = "days";
+    private static final String MONTHS_KEY = "months";
+
+    @Scheduled(cron = "0 0/30 * * * ?")
+    private void runJobs() {
+        logger.debug("Checking "+jobRepo.findByScheduleActiveTrue().size()+" Active Jobs");
+        LocalDateTime schedulerStarted = java.time.LocalDateTime.now(ZoneId.of("UTC"));
+        jobRepo.findByScheduleActiveTrue().forEach(j -> {
+            switch (j.getSchedule().getFrequency()) {
+                case ONCE:
+                    LocalTime jobStartTime = buildJobStartTime(j.getSchedule().getScheduleData().get(START_TIME_KEY));
+                    LocalDateTime jobDate = LocalDateTime.parse(j.getSchedule().getScheduleData().get(DATE_KEY), getDateTimeFormatter());
+
+                    if (jobDate.getDayOfMonth() == schedulerStarted.getDayOfMonth() && jobStartTime.getHour() == schedulerStarted.getHour() && jobStartTime.getMinute() == schedulerStarted.getMinute()) {
+                        startJob(j);
+                        j.getSchedule().setActive(false);
+                        jobRepo.save(j);
+                    }
+                break;
+                case HOURLY:
+                    List<Integer> hourlyDaysToRun = buildDaysToRun(j.getSchedule().getScheduleData().get(DAYS_KEY));
+
+                    if (schedulerStarted.getMinute() == 0 && hourlyDaysToRun.contains(schedulerStarted.getDayOfWeek().getValue())) {
+                        startJob(j);
+                    }
+                break;
+                case DAILY:
+                    List<Integer> daysToRun = buildDaysToRun(j.getSchedule().getScheduleData().get(DAYS_KEY));
+                    LocalTime dailyJobStartTime = buildJobStartTime(j.getSchedule().getScheduleData().get(START_TIME_KEY));
+
+                    if (daysToRun.contains(schedulerStarted.getDayOfWeek().getValue()) && dailyJobStartTime.getHour() == schedulerStarted.getHour() && dailyJobStartTime.getMinute() == schedulerStarted.getMinute()) {
+                        startJob(j);
+                    }
+                break;
+                case MONTHLY:
+                    LocalTime monthlyStartTime = buildJobStartTime(j.getSchedule().getScheduleData().get(START_TIME_KEY));
+                    List<Integer> monthsToRun = buildDaysToRun(j.getSchedule().getScheduleData().get(MONTHS_KEY));
+
+                    if (monthsToRun.contains(schedulerStarted.getMonth().getValue()) && schedulerStarted.getDayOfMonth() == 1 && monthlyStartTime.getHour() == schedulerStarted.getHour() && monthlyStartTime.getMinute() == schedulerStarted.getMinute()) {
+                        startJob(j);
+                    }
+                break;
+            }
+        });
+    }
+
+    private void startJob(Job job) {
+        logger.debug("Starting Job: "+job.getName()+" "+job.getSchedule().getFrequency());
+        processorService.process(job);
```

Comment:
ProcessService process returns boolean if job was started. May want to only use to conditional save the job. Probably only applicable to edge cases.

Reply:
We want to deactivate anyway.

If the job failed to run successfully, it's not because the scheduler failed to start it at the specified time.